### ISSUE TITLE
Podman Machine Date falls behind

### ISSUE THREAD

**Issue by mindovermiles262** [containers/podman#18552]
### Issue Description

When the podman machine is left running, and the host system goes into sleep mode, the clock of the podman machine falls behind until it's so far behind that SSL certificates are invalid.

```
% podman pull hashicorp/terraform
Resolving "hashicorp/terraform" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/hashicorp/terraform:latest...

Error: copying system image from manifest list: parsing image configuration: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/a2/a2a5675db360295f
bf2167e64f7680c186126c7441df7e30c4f52d902474951b/data?verify=1683905416-Jb7bV11yr5KIk4aS%2BM8IY2UH%2Bgc%3D": tls: failed to verify certificate: x509: certificate has expired or is not ye
t valid: current time 2023-05-11T07:06:36-06:00 is before 2023-05-12T00:00:00Z
```

(Note here that the "current time" is 2023-05-11T07:06:36-06:00, when in fact, the current time is 2023-05-12 08:41:20-06:00)

### Steps to reproduce the issue

Steps to reproduce the issue
1. Start Podman Machine
2. Allow computer to enter sleep mode for an extended period of time
3. Pull pod and/or ssh into podman machine and run `date`

### Describe the results you received

Podman machine date was ~1 day behind

### Describe the results you expected

Podman machine date to be accurate

### podman info output

```yaml
% podman version
Client:       Podman Engine
Version:      4.4.2
API Version:  4.4.2
Go Version:   go1.20.1
Git Commit:   74afe26887f814d1c39925a1624851ef3590e79c
Built:        Thu Feb 23 08:59:21 2023
OS/Arch:      darwin/amd64

Server:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.20.2
Built:        Fri Apr 14 09:42:22 2023
OS/Arch:      linux/amd64
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

**Comment by mindovermiles262** [comment]
A quick fix for this is to restart the podman machine:

```
podman machine stop
podman machine start
```

**Comment by arixmkii** [comment]
I think it is believed to be fixed in scope of https://github.com/containers/podman/pull/17661

But this would only work if you created this machine with the latest version of Podman, because there has to be ingnition changes applied during init. 

Has you experienced this with the new blank machine?

**Comment by ashley-cui** [comment]
This is fixed in Podman v4.5.0. You also need to re-create your podman machine with podman 4.5.0 in order for the fix to take effect. Feel free to re-open this issue if the time is sitll not synced after upgrading your version.
